src/asn_ber_der.ml

```diff
@@ -117,13 +120,12 @@ module R = struct
   end
 
   module Gen = struct
-
-    let eof1 cs = cs.Cstruct.len = 0
-    and eof2 cs = Cstruct.LE.get_uint16 cs 0 = 0
+    let eof1 cs = String.length cs = 0
+    and eof2 cs = string_get_uint16_be cs 0 = 0
 
     let split_off cs off n =
       let k = off + n in
-      Cstruct.(sub cs off n, sub cs k (length cs - k))
+      String.sub cs off n, String.sub cs k (String.length cs - k)
```

Comment:
Same here - `Cstruct.sub` and `String.sub` have very different performance profiles. I would be interested to understand a bit more how this function is used (and if performance matters at all)

Reply:
see b421034. for the specific values copying the string is what we do. I'd do this for now, and when we encounter a bottleneck develop some "sstruct" or "bstruct" library, but I don't think this is necessary upfront.

Reply:
I'm fine with doing this that way - maybe just add a comment either in the CHANGES and/or in the docstring of the function to note that this might have change the performance vs. the previous version. It's probably fine, but if anyone hits this, that'll help to have this comment around :-)